pkg/config/config.go

```diff
@@ -238,7 +239,12 @@ func SetEmailConfig(smtpHost string, smtpPort int, senderEmail string, gmailAppP
 }
 
 func GetUIDomain() string {
-	return GetQueryHostname()
+	hostname := GetQueryHostname()
+	parsedUrl, err := url.Parse(hostname)
+	if err != nil {
+		log.Fatalf("GetUIDomain: Failed to parse QueryHostname: %v, err: %v", hostname, err)
```

Comment:
Are you sure we want to crash if we get an error here? This function could feasibly get called at any point, not just at startup

Reply:
I changed this to Errorf and setting it to localhost in case of error